File: Java/src/irrgarten/TestP1.java
///////////////////////////////////////
// Álvaro Maldonado Medina     	     //
// 2D - D3                           //
// Fichero: irrgarten/TestP1.java    //
///////////////////////////////////////


package irrgarten;

import irrgarten.Directions;
import irrgarten.GameCharacter;
import irrgarten.Orientation;
import java.util.Scanner;


public class TestP1 {
    
    public static void main(String[] args) {
        
        
        Dice dado = new Dice();
        int ITERACIONES = 100;
        String salto_linea = "\n";

        // PRUEBA DE LA CLASE WEAPON

        System.out.println("PRUEBA CLASE WEAPON");
        Weapon arma = new Weapon(Dice.weaponPower(), Dice.usesLeft());
        System.out.println(arma.toString());
        arma.attack();
        System.out.println("Ataque realizado");
        System.out.println(arma.toString());
        arma.discard();
        System.out.println(arma.toString());
        
        // PRUEBA CLASE SHIELD
        
        System.out.println("PRUEBA CLASE SHIELD");
        Shield escudo = new Shield(Dice.shieldPower(),Dice.usesLeft());
        System.out.println(escudo.toString()); 
        System.out.println("Escudo realizado");
        escudo.protect();
        System.out.println(escudo.toString());
        
        //Prueba de los tipos Enumerados
        
        Enum orientacion = Orientation.VERTICAL;
        System.out.println("El arma se encuentra en una orientación " +orientacion);
        
        //PRUEBA CLASE DICE
        
        System.out.println(salto_linea);
        System.out.println("MÉTODO randomPos");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.randomPos(10));
        }
        
        System.out.println(salto_linea);
        System.out.println("MÉTODO whoStarts");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.whoStarts(10));
        }
        
        System.out.println(salto_linea);
        System.out.println("Método randomIntelligence");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.randomIntelligence());
        }
        
        System.out.println(salto_linea);
        System.out.println("Método randomStrength");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.randomStrength());
        }
        
        System.out.println(salto_linea);
        System.out.println("Método resurrectPlayer");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.resurrectPlayer());
        }
        
        System.out.println(salto_linea);
        System.out.println("Método weaponsReward");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.weaponsReward());
        }
        
        System.out.println(salto_linea);
        System.out.println("Método shieldsReward");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.shieldsReward());
        }

        System.out.println(salto_linea);
        System.out.println("Método healthReward");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.healthReward());
        }
        
        System.out.println(salto_linea);
        System.out.println("Método weaponPower");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.weaponPower());
        }
        
        System.out.println(salto_linea);
        System.out.println("Método shieldPower");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.shieldPower());
        }
        
        System.out.println(salto_linea);
        System.out.println("Método usesLeft");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.usesLeft());
        }
        
        System.out.println(salto_linea);
        System.out.println("Método intensity");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.intensity(10));
        }
        
        System.out.println(salto_linea);
        System.out.println("Método discardElement");
        
        for(int i = 0; i < ITERACIONES; i++){
            System.out.println(dado.discardElement(dado.usesLeft()));
        }
        
        System.out.println(salto_linea);


    }

}

File: Ruby/TestP1.rb
#encoding:utf-8

####################################### 
# Álvaro Maldonado Medina             #      
# 2D- D3                              #
# Fichero: Irrgarten/TestP1.rb        #
#######################################


module Irrgarten

    require_relative 'Dice.rb'
    require_relative 'Orientation.rb'
    require_relative 'Directions.rb'
    require_relative 'GameCharacter.rb'
    require_relative 'GameState.rb'
    require_relative 'Shield.rb'
    require_relative 'Weapon.rb'

end

def main

    def separation_lines
        puts "===================================================================="
    end

    def salto_linea
        puts "\n"
    end


    dado = Irrgarten::Dice.new()
    n_decimales = 3     # nº máximo de decimales en coma flotante a mostrar
    iteraciones = 100   # nº de iteraciones que se nos pide en el guion por 
                        # método de la clase Dice




    #PRUEBA DE LA CLASE WEAPON

    separation_lines
    puts "PROBANDO CLASE WEAPON..."

    arma = Irrgarten::Weapon.new(dado.MAX_ATTACK.to_f, Dice.MAX_USES)

    #Prueba      

    print "Método to_s: "
    puts arma.to_s
    print "Realizando ataque..."
    arma.attack
    puts arma.to_s


    #PRUEBA DE LA CLASE SHIELD

    separation_lines
    puts "PROBANDO CLASE SHIELD"


    escudo = Irrgarten::Shield.new(dado.get_max_shield.to_f, dado.get_max_uses)

    #Prueba    

    puts "Método to_s"
    puts escudo.to_s
    puts "Realizando acción de protección..."
    escudo.protect
    puts escudo.to_s

    #PRUEBA DE LA CLASE DICE (100 iteraciones por cada método)

    separation_lines
    puts "PROBANDO CLASE DICE"


    salto_linea
    puts "Método randomPos para 10 filas o columnas" 

    for i in 0..(iteraciones)
        puts dado.random_pos(10)
    end
    salto_linea


    puts "Método whoStarts para 16 jugadores (0-15)"

    for i in 0..(iteraciones)
        puts dado.who_starts(15)
    end
    salto_linea


    puts "Método randomIntelligence"

    for i in 0..(iteraciones)
        puts dado.random_intelligence.round(n_decimales)
    end
    salto_linea


    puts "Método randomStrength"

    for i in 0..(iteraciones)
        puts dado.random_strength.round(n_decimales)
    end


    salto_linea
    puts "Método resurrectPlayer"

    for i in 0..(iteraciones)
        puts dado.resurrect_player
    end


    salto_linea
    puts "Método weaponsReward"

    for i in 0..(iteraciones)
        puts dado.weapons_reward
    end


    salto_linea
    puts "Método shieldsReward"

    for i in 0..(iteraciones)
        puts dado.shields_reward
    end


    salto_linea
    puts "Método healthReward"

    for i in 0..(iteraciones)
        puts dado.health_reward
    end


    salto_linea
    puts "Método weaponPower"

    for i in 0..(iteraciones)
        puts dado.weapon_power.round(n_decimales)
    end


    salto_linea
    puts "Método shieldPower"

    for i in 0..(iteraciones)
        puts dado.shield_power.round(n_decimales)
    end


    salto_linea
    puts "Método usesLeft"

    for i in 0..(iteraciones)
        puts dado.uses_left
    end


    salto_linea
    puts "Método intensity"

    for i in 0..(iteraciones)
        puts dado.intensity(10).round(n_decimales)
    end


    salto_linea
    puts "Método discardElement"

    for i in 0..(iteraciones)
        puts dado.discardElement(dado.get_max_uses)
    end

end

main
